src/pages/VeSDL/VoteHistory.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Button,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+  styled,
+  useTheme,
+} from "@mui/material"
+import React, { useCallback, useEffect, useState } from "react"
+import { BigNumber } from "ethers"
+import { GaugeController } from "../../../types/ethers-contracts/GaugeController"
+import { GaugeName } from "./OnChainVote"
+import { enqueuePromiseToast } from "../../components/Toastify"
+import { useActiveWeb3React } from "../../hooks"
+
+const VoteTableRow = styled(TableRow)(({ theme }) => ({
+  "td:first-child": {
+    borderTopLeftRadius: "10px",
+    borderBottomLeftRadius: "10px",
+  },
+  "td:last-child": {
+    borderTopRightRadius: "10px",
+    borderBottomRightRadius: "10px",
+    padding: 0,
+  },
+  backgroundColor: theme.palette.background.paper,
+}))
+
+type VoteList = {
+  gaugeAddress: string
+  gaugeName: string
+  weight: BigNumber
+}
+
+type Props = {
+  gaugeControllerContract: GaugeController | null
+  gaugeNames: (GaugeName | undefined)[]
+}
+
+export default function VoteHistory({
+  gaugeControllerContract,
+  gaugeNames,
+}: Props) {
+  const theme = useTheme()
+  const { account } = useActiveWeb3React()
+  const [voteUsed, setVoteUsed] = useState<number | undefined>()
```

Comment:
You are right.

Reply:
If we set vote used as `0`, then it will show `0` while fetching data from contract. It is good to remove `undefined` from value but this value is not sued on other function, so I think, it is rather to hide this value when there is no vote used value.